src/pudl/dbt_wrapper.py

```diff
@@ -86,3 +88,40 @@ def build_with_context(
     failure_contexts = get_context_for_nodes(failed_nodes)
 
     return BuildResult(success=build_output.success, failure_contexts=failure_contexts)
+
+
+def dagster_to_dbt_selection(
+    selection: str, defs: dg.Definitions | None = None, manifest=None
+) -> str:
+    """Translate dagster asset selection to db node selection.
+
+    * turn asset selection into asset keys
+    * turn asset keys into node names
+    * turn node names into selection string
+    """
+    if defs is None:
+        import pudl.etl
+
+        defs = pudl.etl.defs
+
+    asset_keys = dg.AssetSelection.from_string(selection).resolve(
+        defs.resolve_asset_graph()
+    )
+    asset_names = {asset_key.to_user_string() for asset_key in asset_keys}
+
+    if manifest is None:
+        manifest_path = PUDL_ROOT_PATH / "dbt" / "target" / "manifest.json"
+        with chdir(PUDL_ROOT_PATH / "dbt"):
```

Comment:
It can be a little slow to run `dbt parse` every time, but if we want to make sure the manifest is up-to-date and we're not dropping any tests we have to do it. Or, we can go and check to see if the manifest is fresher than all the schemas, but that seems fragile so I decided against it for now.